### PR TITLE
[ENG-4182] Add moderator collection moderation endpoints

### PR DESCRIPTION
**Pull request by Johnetordoff**
## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

https://openscience.atlassian.net/jira/software/c/projects/ENG/issues/ENG-4182

### CODE REVIEW

File: osf_tests/test_schema_responses.py [hunk 1]
```diff
@@ -32,21 +31,6 @@
     'q1': '', 'q2': '', 'q3': '', 'q4': [], 'q5': '', 'q6': []
 }
 
-def _ensure_subscriptions(provider):
```
Comment (Johnetordoff) [inline osf_tests/test_schema_responses.py:35]: Just moving this to a shared folder, not just for schema responses.

File: api_tests/providers/collections/views/test_collections_provider_moderator_list.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
+import mock
```
Comment (Johnetordoff) [inline api_tests/providers/collections/views/test_collections_provider_moderator_list.py:1]: This is mostly copied form other provider moderator tests with a few updates.

File: api_tests/providers/collections/views/test_collections_provider_moderator_list.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        data['data']['id'] = user_id
+    if email:
+        data['data']['attributes']['email'] = email
+    return data
+
+
+@pytest.mark.django_db
+class TestGETCollectionsModeratorList:
+
+    def test_GET_unauthorized(self, app, url, provider):
```
Comment (jwalz) [inline api_tests/providers/collections/views/test_collections_provider_moderator_list.py:66]: nit: technically unauthenticated, not unauthorized

File: framework/auth/oauth_scopes.py [hunk 4]
```diff
@@ -257,8 +264,8 @@ class ComposedScopes(object):
 
     # Organizer Collections collection
     # Using Organizer Collections and the node links they collect. Reads Node Metadata.
-    ORGANIZER_READ = (CoreScopes.ORGANIZER_COLLECTIONS_BASE_READ, CoreScopes.COLLECTED_META_READ,) + NODE_METADATA_READ
-    ORGANIZER_WRITE = ORGANIZER_READ + (CoreScopes.ORGANIZER_COLLECTIONS_BASE_WRITE, CoreScopes.NODE_LINKS_WRITE, CoreScopes.COLLECTED_META_WRITE)
+    ORGANIZER_READ = (CoreScopes.ORGANIZER_COLLECTIONS_BASE_READ, CoreScopes.READ_COLLECTION_SUBMISSION,) + NODE_METADATA_READ
+    ORGANIZER_WRITE = ORGANIZER_READ + (CoreScopes.ORGANIZER_COLLECTIONS_BASE_WRITE, CoreScopes.NODE_LINKS_WRITE, CoreScopes.WRITE_COLLECTION_SUBMISSION)
```
Comment (jwalz) [inline framework/auth/oauth_scopes.py:268]: If this is a replacement for `COLLECTED_META_READ`, make sure you fix existing API Endpoints

File: framework/auth/oauth_scopes.py [hunk 5]
```diff
@@ -192,6 +192,13 @@ class CoreScopes(object):
     READ_REGISTRATION_RESOURCES = 'read_registration_resources'
     WRITE_REGISTRATION_RESOURCES = 'write_registration_resources'
 
+    READ_COLLECTION_SUBMISSION_ACTION = 'read_collection_submission_action'
+    WRITE_COLLECTION_SUBMISSION_ACTION = 'write_collection_submission_action'
```
Comment (jwalz) [inline framework/auth/oauth_scopes.py:196]: I don't think we need these. We use `[READ/WRITE]_SCHEMA_RESPONSE` as our scope for `SchemaResponseActionList`, for example.

File: api/providers/serializers.py [hunk 6]
```diff
@@ -382,3 +383,21 @@ def get_absolute_url(self, obj):
                 'version': self.context['request'].parser_context['kwargs']['version'],
             },
         )
+
+
+class CollectionsModeratorSerializer(ModeratorSerializer):
+
+    provider = RegistrationProviderRelationshipField(
```
[the remark this answers is not in the document]
Comment (jwalz) [reply api/providers/serializers.py:390]: (whoops, never submitted this review from last week)

**Review by jwalz — CHANGES_REQUESTED** [review]
Gonna be some conflicts to clean up between this and the other CollectionSubmissionAction PRs, but looks mostly good. Just a couple weird things with CoreScopes

**Pull request merged**